components/script/dom/webglrenderingcontext.rs

```diff
@@ -2015,14 +2061,36 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
         if attrib_id > self.limits.max_vertex_attribs {
             return self.webgl_error(InvalidValue);
         }
+        if stride < 0 || stride > 255 || offset < 0 {
```

Comment:
please, back this with quotes to the spec, such as:

> From GLES 2.0: GL_INVALID_VALUE is generated if stride is negative.
> From WebGL: If offset is negative, an INVALID_VALUE error will be generated. [...] In WebGL, the maximum supported stride is 255.

Ideally always, but specially when the checks belong to different specs.